src/Modules/Admin/Commands.Admin/GetPowerBIWorkspaceEncryptionStatus.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.PowerBI.Common.Api;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Admin
+{
+    [Cmdlet(CmdletVerb, CmdletName)]
+    public class GetPowerBIWorkspaceEncryptionStatus : PowerBIClientCmdlet
+    {
+        public const string CmdletName = "PowerBIWorkspaceEncryptionStatus";
+        public const string CmdletVerb = VerbsCommon.Get;
+
+        public GetPowerBIWorkspaceEncryptionStatus() : base() { }
+
+        public GetPowerBIWorkspaceEncryptionStatus(IPowerBIClientCmdletInitFactory init) : base(init) { }
+
+        #region Parameters
+
+        [Parameter(Mandatory = true)]
+        public string Name { get; set; }
+
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            using (var client = this.CreateClient())
+            {
+                try
+                {
+                    var workspaces = this.GetWorkspaces(client);
+                    var matchedWorkspace = this.GetMatchingWorkspace(workspaces);
```

Comment:
It looks like GetWorkspaces does a filter match, why is GetMatchingWorkspaces also needed as workspaces have to have unique names?

Reply:
It is not needed anymore. Updated the code